packages/utils/src/utils/date-time.ts

```diff
@@ -0,0 +1,212 @@
+/**
+ * Tai64 timestamp.
+ *
+ * @hidden
+ */
+export type Tai64Timestamp = string;
+
+/**
+ * Interface for the DateTime class.
+ */
+export interface IDateTime extends Date {
```

Comment:
There is no need to create an interface if there's only one implementer of it (`DateTime`). Wherever we need just the type of the class, we can do `import type { DateTime } from '...'`, and if we need the actual class we do the standard `import { DateTime } from '...'`. I'm requesting changes for the deletion of the interface.

Reply:
Btw the rest seems good! :+1:

Reply:
> There is no need to create an interface if there's only one implementer of it

I disagree with this as there may be times where something is designed with extensibility in mind, but right now there may only be one implementer. Giving less overhead to the extension.

However context is important, and I agree here the interface is unnecessary. We're in the realm of `AbstractAddress`.

Reply:
[23778c70a32127e963c7dfb2e98749bb42901fdb](https://github.com/FuelLabs/fuels-ts/pull/1627/commits/23778c70a32127e963c7dfb2e98749bb42901fdb)